src/main/kotlin/day3/Cleric.kt

```
@@ -1,30 +1,51 @@
+    val name: String = "",
+    var hp: Int = MAX_HP,
+    var mp: Int = MAX_MP
 ) {
+    companion object {
+        const val MAX_HP = 50
+        const val MAX_MP = 10
+    }
+
+    init {
+        if (name.isEmpty()) {
+            throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)